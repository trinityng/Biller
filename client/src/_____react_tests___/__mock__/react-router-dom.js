import React from 'react';
// Just render plain div with its children
const rrd = require('react-router-dom');

rrd.BrowserRouter = ({children}) => <div>{children}</div>
module.exports = rrd;