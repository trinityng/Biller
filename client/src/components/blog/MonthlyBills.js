import React from 'react';

import { Row, Col, Card, CardHeader, CardBody, Button } from 'shards-react';

import RangeDatePicker from '../common/RangeDatePicker';
import Chart from '../../utils/chart';

class MonthlyBills extends React.Component {
  constructor(props) {
    super(props);
    this.canvasRef = React.createRef();
  }

  renderChart = () => {
    const MonthlyExpenses = new Chart(this.canvasRef.current, {
      type: 'LineWithLine',
      data: this.props.data,
      options: this.props.chartOptions,
    });
    if (this.props.data.datasets > 1) {
      // They can still be triggered on hover.
      const buoMeta = MonthlyExpenses.getDatasetMeta(0);

      buoMeta.data[0]._model.radius = 0;
      buoMeta.data[
        this.props.data.datasets[0].data.length - 1
      ]._model.radius = 0;

      // Render the chart.
      MonthlyExpenses.render();
    }
  };

  componentDidUpdate() {
    this.renderChart();
  }

  componentDidMount() {
    this.renderChart();
  }

  render() {
    return (
      <Card small className="h-100">
        <CardHeader className="border-bottom">
          <h6 className="m-0">Monthly Expenses</h6>
        </CardHeader>
        <CardBody className="pt-0">
          <Row className="border-bottom py-2 bg-light">
            <Col sm="6" className="d-flex mb-2 mb-sm-0">
              <RangeDatePicker />
            </Col>
            <Col>
              <Button
                size="sm"
                className="d-flex btn-white ml-auto mr-auto ml-sm-auto mr-sm-0 mt-3 mt-sm-0"
              >
                View Full Report &rarr;
              </Button>
            </Col>
          </Row>
          <canvas
            height="120"
            ref={this.canvasRef}
            style={{ maxWidth: '100% !important' }}
          />
        </CardBody>
      </Card>
    );
  }
}

export default MonthlyBills;
