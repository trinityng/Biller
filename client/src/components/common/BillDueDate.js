import React from 'react';
import {
  Card,
  CardHeader,
  CardBody,
  ListGroup,
  ListGroupItem,
  CardFooter,
  Row,
  Col,
  Button,
} from 'shards-react';
import { DatePicker } from 'antd';
import moment from 'moment';
import update from 'immutability-helper';

const dateFormat = 'MM/DD';


class BillDueDate extends React.Component {
  constructor(){
    super();

    this.state = {
      title: 'Bill Due Dates',
      accountDueDates: [ 
        {
          name: 'Chase Test Credit',
          key: null,
          date: '12/2',
        },
        {
          name: 'Chase Test Credit VIP',
          key: null,
          date: '5/2',
        },
        {
          name: 'PG&E Test',
          key: null,
          date: '6/2',
        },
      ],
    }

    this.onDateChange = this.onDateChange.bind(this);
    this.onDateUpdate = this.onDateUpdate.bind(this);
  }

  componentDidMount () {

  }

  onDateChange = (index, dateString) => {
    console.log(index + " " + dateString);
    this.setState(update(this.state.accountDueDates[index], {date: {$set: dateString }}));
  }

  onDateUpdate = () => {
    console.log(this.state.accountDueDates);
  }

  render() { 
    return (
      <Card small>
        <CardHeader className="border-bottom">
          <h6 className="m-0">{this.state.title}</h6>
          <div className="block-handle" />
        </CardHeader>

        <CardBody className="p-0">
          <ListGroup small flush className="list-group-small">
            {this.state.accountDueDates.map((item, idx) => (
              <ListGroupItem key={idx} className="d-flex px-3">
                <span className="text-semibold text-fiord-blue">{item.name}</span>
                <span className="ml-auto text-right text-semibold text-reagent-gray">
                  <DatePicker 
                    style = {{ width: 100 }}
                    format = { dateFormat }
                    defaultValue = { moment(item.date, dateFormat) }       
                    allowClear = { false }
                    onChange = {( _dateObj , datestring) => this.onDateChange(idx, datestring)}
                  />
                </span>
              </ListGroupItem>
            ))}
          </ListGroup>
        </CardBody>

        <CardFooter className="border-top">
          <Row>
            {/* Update Button */}
            <Col>
              <Button theme="primary" className="mb-2 mr-1" onClick = {this.onDateUpdate}>
                Update
              </Button>
            </Col>

            {/* View All Due Dates */}
            <Col className="text-right view-report">
              {/* eslint-disable-next-line */}
              <a href="#">See all due dates &rarr;</a>
            </Col>
          </Row>
        </CardFooter>
      </Card>
    );  
  }
};

export default BillDueDate;
