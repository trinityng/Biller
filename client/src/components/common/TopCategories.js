import React from 'react';
import PropTypes from 'prop-types';
import {
  Card,
  CardHeader,
  CardBody,
  ListGroup,
  ListGroupItem,
  CardFooter,
  Row,
  Col,
  FormSelect,
} from 'shards-react';

const TopCategories = props => (
  <Card small>
    <CardHeader className="border-bottom">
      <h6 className="m-0">Categories</h6>
      <div className="block-handle" />
    </CardHeader>

    <CardBody className="p-0">
      <ListGroup small flush className="list-group-small">
        {props.category_expense.map((item, idx) => (
          <ListGroupItem key={idx} className="d-flex px-3">
            <span className="text-semibold text-fiord-blue">
              {item.category}
            </span>
            <span className="ml-auto text-right text-semibold text-reagent-gray">
              ${item.total.toFixed(0)}
            </span>
          </ListGroupItem>
        ))}
      </ListGroup>
    </CardBody>

    <CardFooter className="border-top">
      <Row>
        {/* Time Span */}
        <Col>
          <FormSelect
            size="sm"
            value="last-week"
            style={{ maxWidth: '130px' }}
            onChange={() => {}}
          >
            <option value="last-week">Last Week</option>
            <option value="today">Today</option>
            <option value="last-month">Last Month</option>
            <option value="last-year">Last Year</option>
          </FormSelect>
        </Col>

        {/* View Full Report */}
        <Col className="text-right view-report">
          {/* eslint-disable-next-line */}
          <a href="#">Full report &rarr;</a>
        </Col>
      </Row>
    </CardFooter>
  </Card>
);

TopCategories.propTypes = {
  /**
   * The component's title.
   */
  title: PropTypes.string,
  /**
   * The referral data.
   */
  referralData: PropTypes.array,
};

TopCategories.defaultProps = {
  title: 'Top Categories',
  referralData: [
    {
      title: 'Shopping',
      value: '$1,291',
    },
    {
      title: 'Travel',
      value: '$1,291',
    },
    {
      title: 'Food',
      value: '$1,291',
    },
    {
      title: 'Utilities',
      value: '$1,291',
    },
  ],
};

export default TopCategories;
