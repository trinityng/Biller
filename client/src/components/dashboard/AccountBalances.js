import React from 'react';
import { Card, CardHeader, CardBody } from 'shards-react';

import Chart from '../../utils/chart';

class AccountBalances extends React.Component {
  constructor(props) {
    super(props);

    this.canvasRef = React.createRef();
  }

  renderChart() {
    const chartConfig = {
      type: 'pie',
      data: this.props.chartData,
      options: {
        ...{
          legend: {
            position: 'bottom',
            labels: {
              padding: 25,
              boxWidth: 20,
            },
          },
          cutoutPercentage: 0,
          tooltips: {
            custom: false,
            mode: 'index',
            position: 'nearest',
          },
        },
        ...this.props.chartOptions,
      },
    };

    new Chart(this.canvasRef.current, chartConfig);
  }

  componentDidUpdate() {
    this.renderChart();
  }

  componentDidMount() {
    this.renderChart();
  }

  render() {
    return (
      <Card small className="h-100">
        <CardHeader className="border-bottom">
          <h6 className="m-0">Account Balances</h6>
        </CardHeader>
        <CardBody className="d-flex py-0">
          <canvas
            height="220"
            ref={this.canvasRef}
            className="blog-users-by-device m-auto"
          />
        </CardBody>
      </Card>
    );
  }
}

export default AccountBalances;
