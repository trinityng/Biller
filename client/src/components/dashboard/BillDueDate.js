import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  Card,
  CardHeader,
  CardBody,
  ListGroup,
  ListGroupItem,
  CardFooter,
  Badge,
  Row,
  Col,
} from 'shards-react';
import { DatePicker } from 'antd';
import moment from 'moment';
import { USER_TOKEN } from '../../constants';
import {
  changeBillDueDate,
  plaidBills,
  getNotification,
} from '../../api/plaid.api';

const dateFormat = 'MM-DD-YY';

class BillDueDate extends React.Component {
  onDateChange = (item, value) => {
    const token = localStorage.getItem(USER_TOKEN);
    const { email } = this.props.user;
    const { name } = item;

    const date =
      moment()
        .format('YYYY')
        .toString() +
      '-' +
      value.format('MM-DD').toString();

    this.props
      .changeBillDueDate({
        account_name: name,
        due_date: date,
        email: email,
        token: token,
      })
      .then(() => {
        const token = localStorage.getItem(USER_TOKEN);
        const user = { token, email };
        this.props.plaidBills(user);
        this.props.getNotification(user);
      });
  };

  render() {
    const { plaid } = this.props || {};
    const { bills } = plaid || [];

    return (
      <Card small className="h-100">
        <CardHeader className="border-bottom">
          <h6 className="m-0">Bill Due Dates</h6>
          <div className="block-handle" />
        </CardHeader>

        <CardBody className="p-0">
          <ListGroup small flush className="list-group-small">
            {bills && bills.length ? (
              bills.map((item, idx) => (
                <ListGroupItem key={idx} className="d-flex px-3">
                  <span className="text-semibold text-fiord-blue">
                    {item.name}
                    <br />
                    Amount:{' '}
                    {item.amount > 0 ? (
                      <Badge theme="danger" className="mb-0 mr-0">
                        ${item.amount}
                      </Badge>
                    ) : item.amount === 0 ? (
                      <Badge theme="success" className="mb-0 mr-0">
                        ${item.amount}
                      </Badge>
                    ) : (
                      <Badge theme="success" className="mb-0 mr-0">
                        -${item.amount * -1}
                      </Badge>
                    )}
                  </span>
                  <span className="ml-auto text-right text-semibold text-reagent-gray">
                    {item.due_date ? (
                      <DatePicker
                        showTime={{ format: 'HH:mm' }}
                        style={{ width: 120 }}
                        format={dateFormat}
                        defaultValue={moment(
                          moment(item.due_date, 'YYYY-MM-DD')
                            .format(dateFormat)
                            .toString(),
                          dateFormat
                        )}
                        allowClear={false}
                        disabled={item.amount <= 0 ? true : false}
                        onChange={this.onChange}
                        onOk={(item, value => this.onDateChange(item, value))}
                      />
                    ) : (
                      <DatePicker
                        showTime={{ format: 'HH:mm' }}
                        style={{ width: 120 }}
                        format={dateFormat}
                        allowClear={false}
                        disabled={item.amount <= 0 ? true : false}
                        onOk={(item, value => this.onDateChange(item, value))}
                      />
                    )}
                  </span>
                </ListGroupItem>
              ))
            ) : (
              <span />
            )}
          </ListGroup>
        </CardBody>

        <CardFooter className="border-top">
          <Row>
            {/* View All Due Dates */}
            <Col className="text-right view-report">
              {/* eslint-disable-next-line */}
              <a href="#">See all due dates &rarr;</a>
            </Col>
          </Row>
        </CardFooter>
      </Card>
    );
  }
}

// Store
function mapStateToProps(state) {
  return {
    user: state.user,
    plaid: state.plaid,
  };
}

function matchDispatchToProps(dispatch) {
  return bindActionCreators(
    { plaidBills, changeBillDueDate, getNotification },
    dispatch
  );
}

export default connect(mapStateToProps, matchDispatchToProps)(BillDueDate);
