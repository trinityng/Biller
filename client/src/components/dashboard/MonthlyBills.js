import React from 'react';
import { Card, CardHeader, CardBody } from 'shards-react';
import Chart from '../../utils/chart';

class MonthlyBills extends React.Component {
  constructor(props) {
    super(props);
    this.canvasRef = React.createRef();
  }

  renderChart = () => {
    const MonthlyExpenses = new Chart(this.canvasRef.current, {
      type: 'LineWithLine',
      data: this.props.data,
      options: this.props.chartOptions,
    });
    if (this.props.data.datasets > 1) {
      // They can still be triggered on hover.
      const buoMeta = MonthlyExpenses.getDatasetMeta(0);

      buoMeta.data[0]._model.radius = 0;
      buoMeta.data[
        this.props.data.datasets[0].data.length - 1
      ]._model.radius = 0;

      // Render the chart.
      MonthlyExpenses.render();
    }
  };

  componentDidUpdate() {
    this.renderChart();
  }

  componentDidMount() {
    this.renderChart();
  }

  render() {
    return (
      <Card small className="h-100">
        <CardHeader className="border-bottom">
          <h6 className="m-0">Monthly Expenses</h6>
        </CardHeader>
        <CardBody className="pt-0">
          <canvas
            height="120"
            ref={this.canvasRef}
            style={{ maxWidth: '100% !important' }}
          />
        </CardBody>
      </Card>
    );
  }
}

export default MonthlyBills;
