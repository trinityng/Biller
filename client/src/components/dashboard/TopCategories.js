import React from 'react';
import PropTypes from 'prop-types';
import {
  Card,
  CardHeader,
  CardBody,
  ListGroup,
  ListGroupItem,
  Badge,
} from 'shards-react';

const TopCategories = props => (
  <Card small>
    <CardHeader className="border-bottom">
      <h6 className="m-0">Categories</h6>
      <div className="block-handle" />
    </CardHeader>

    <CardBody className="p-0">
      <ListGroup small flush className="list-group-small">
        {props.category_expense.map((item, idx) => (
          <ListGroupItem key={idx} className="d-flex px-3">
            <span className="text-semibold text-fiord-blue">
              {item.category}
            </span>
            <Badge
              theme="info"
              className="mb-0 mr-0 ml-auto text-right text-semibold "
            >
              ${item.total.toFixed(0)}
            </Badge>
          </ListGroupItem>
        ))}
      </ListGroup>
    </CardBody>
  </Card>
);

TopCategories.propTypes = {
  /**
   * The component's title.
   */
  title: PropTypes.string,
  /**
   * The referral data.
   */
  referralData: PropTypes.array,
};

TopCategories.defaultProps = {
  title: 'Top Categories',
  referralData: [
    {
      title: 'Shopping',
      value: '$1,291',
    },
    {
      title: 'Travel',
      value: '$1,291',
    },
    {
      title: 'Food',
      value: '$1,291',
    },
    {
      title: 'Utilities',
      value: '$1,291',
    },
  ],
};

export default TopCategories;
