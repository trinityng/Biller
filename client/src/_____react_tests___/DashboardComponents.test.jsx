import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {Provider} from 'react-redux';
import {mount, shallow, configure } from 'enzyme';
import { MemoryRouter, Router } from "react-router-dom";
import React from 'react';
import {Input, Button} from 'antd';
import Adapter from 'enzyme-adapter-react-16';

import PageTitle from '../components/common/PageTitle';
import SmallStats from '../components/common/SmallStats';
import MonthlyBills from '../components/dashboard/MonthlyBills';
import AccountBalances from '../components/dashboard/AccountBalances';
import TopCategories from '../components/dashboard/TopCategories';
import BillDueDate from '../components/dashboard/BillDueDate';


Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

configure({adapter: new Adapter()});
/*
describe('dashboard components', () => {
  let initialState = {};
  let store;

  beforeEach(() => {
    store = mockStore(initialState)
  });

  describe("when component is mounted", () => {
    beforeEach(() => {
        store = mockStore(initialState)
    });

    it(' should render PageTitle correctly', () => {
        const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter initialEntries={['/dashboard']}>
                <PageTitle />
            </MemoryRouter>,
        </Provider>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it(' should render MonthlyBills correctly', () => {
      const wrapper = mount(
      <Provider store={store}>
          <MemoryRouter initialEntries={['/dashboard']}>
              <MonthlyBills />
          </MemoryRouter>,
      </Provider>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it(' should render MonthlyBills correctly', () => {
      const wrapper = mount(
      <Provider store={store}>
          <MemoryRouter initialEntries={['/dashboard']}>
              <div />
          </MemoryRouter>,
      </Provider>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it(' should render AccountBalances correctly', () => {
      const wrapper = mount(
      <Provider store={store}>
          <MemoryRouter initialEntries={['/dashboard']}>
              <AccountBalances />
          </MemoryRouter>,
      </Provider>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it(' should render TopCategories correctly', () => {
      const wrapper = mount(
      <Provider store={store}>
          <MemoryRouter initialEntries={['/dashboard']}>
              <TopCategories />
          </MemoryRouter>,
      </Provider>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it(' should render BillDueDate correctly', () => {
      const wrapper = mount(
      <Provider store={store}>
          <MemoryRouter initialEntries={['/dashboard']}>
              <BillDueDate />
          </MemoryRouter>,
      </Provider>
      );

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when user is logged in", () => {
    it('should display correct user bank data', () => {
     
    });
    it('should graph correct bank data', () => {
     
    });
  });

  describe("when user interacts with component", () => {
    it('should work when user change dates', () => {
     
    });
    it('should work when user goes to other pages', () => {
     
    });
  });
});
*/

describe('test', () => {
  let initialState = {};
  let store;

  beforeEach(() => {
    store = mockStore(initialState)
  });

  it('should render correctly', () => {
    const wrapper = mount(
        <div />
    );

    expect(wrapper).toMatchSnapshot();
  });
});