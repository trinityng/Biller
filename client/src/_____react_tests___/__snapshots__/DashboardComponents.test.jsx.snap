// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should render correctly 1`] = `ReactWrapper {}`;
